pkg/workqueue/inmem/inmem.go

```diff
@@ -173,9 +175,10 @@ func (o *inProgressKey) Context() context.Context {
 }
 
 type queuedKey struct {
-	wq       *wq
-	key      string
-	priority int64
+	workqueue.Options
```

Comment:
As I added more fields, this felt like a much cleaner way of passing this stuff around for in-memory.